src/interface/console.c

```diff
@@ -459,47 +490,52 @@ static int cc_rides(const utf8 **argv, int argc)
 			if (argc < 4) {
 				console_printf("rides set type <ride id> <ride type>");
 				console_printf("rides set friction <ride id> <friction value>");
+				console_printf("rides set powered_velocity <ride id> <velocity> [acceleration]");
 				return 0;
 			}
+			int optional;
+			bool int_valid[3] = { 0 };
+			int ride_index = console_parse_int(argv[2], &int_valid[0]);
+			int value = console_parse_int(argv[3], &int_valid[1]);
+			if (argc > 4) {
+				optional = console_parse_int(argv[4], &int_valid[2]);
+			}
+			if (!int_valid[0] || !int_valid[1] || ((argc > 4) && !int_valid[2])) {
+				console_printf("This command expects integer arguments");
+				return 0;
+			}
+			if ((ride_index < 0) || (ride_index > 254)) {
+				console_printf("Ride index must be between 0 and 254");
+				return 0;
+			}
+			rct_ride *ride = get_ride(ride_index);
+			if (ride->type == RIDE_TYPE_NULL) {
+				console_printf("No ride found with index %d", ride_index);
+				return 0;
+			}
+
+
 			if (strcmp(argv[1], "type") == 0) {
-				bool int_valid[3] = { 0 };
-				int ride_index = console_parse_int(argv[2], &int_valid[0]);
-				int type = console_parse_int(argv[3], &int_valid[1]);
-				if (!int_valid[0] || !int_valid[1]) {
-					console_printf("This command expects integer arguments");
-				} else if (ride_index < 0) {
-					console_printf("Ride index must not be negative");
-				} else {
-					gGameCommandErrorTitle = STR_CANT_CHANGE_OPERATING_MODE;
-					int res = game_do_command(0, (type << 8) | 1, 0, (RIDE_SETTING_RIDE_TYPE << 8) | ride_index, GAME_COMMAND_SET_RIDE_SETTING, 0, 0);
-					if (res == MONEY32_UNDEFINED) {
-						console_printf("That didn't work");
-					}
+				gGameCommandErrorTitle = STR_CANT_CHANGE_OPERATING_MODE;
+				int res = game_do_command(0, (value << 8) | 1, 0, (RIDE_SETTING_RIDE_TYPE << 8) | ride_index, GAME_COMMAND_SET_RIDE_SETTING, 0, 0);
+				if (res == MONEY32_UNDEFINED) {
+					console_printf("That didn't work");
```

Comment:
"Could not set the velocity or acceleration for this vehicle" or something along these lines.